### ISSUE TITLE
DVC could use the unix SSH client config files

### ISSUE THREAD

**Issue by PeterFogh** [iterative/dvc#1613]
Hi, 
The reasoning for this feature request is:
*In a team where all members uses SSH to a compute server from individual machines with individual credentials, it is sometimes normal practice to use the SSH client config file to define the host settings. See [this website](https://www.digitalocean.com/community/tutorials/how-to-configure-custom-connection-options-for-your-ssh-client) for examples.*

Thus, my feature request is: 
**Can DVC fallback to use the SSH client config files when a DVC remote of the specified name is not defined in the DVC config files OR can the user simply configure the DVC repository to use the SSH client config files instead of the DVC SSH remotes.**

--------------------------------------------------------
**Do you need a fix now - here it is.**
I have come to know that DVC also can define this individual credentials per machines (i.e. team member). DVC do this by configuring a global SSH remote, for example by the following indevidual commands:
```
dvc remote add compute_server ssh://[IP-address_or_domain-name]/ --global
dvc remote modify compute_server user [SSH-server-username] --global
dvc remote modify compute_server port [SSH-port-at-SSH-server] --global
dvc remote modify compute_server keyfile [Path-to-local-private-SSH-key] --global
```
After this global SSH remove is defined by each team member on their local machine, everybody in the team can create a DVC pipeline step with [external dependecies](https://dvc.org/doc/user-guide/external-dependencies) for [outputs](https://dvc.org/doc/user-guide/external-outputs), for example:
`dvc run -d ssh://compute_server/home/[username]/classify/data/Posts.xml -d code/xml_to_tsv.py -d code/conf.py -o ssh://compute_server/home/[username]/classify/data/Posts.tsv python code/xml_to_tsv.py`
And the beautiful thing is that every team member can reproduce this pipeline without SSH credential problems :)

**Comment by efiop** [comment]
For the record: Another user on discord was using these configs https://discordapp.com/channels/485586884165107732/485596304961962003/573519544153538566 .

**Comment by PeterFogh** [comment]
@khamutov Thanks for implementing my feature request :) Looking forward to trying it out.

**Comment by efiop** [comment]
@khamutov @PeterFogh Released in 0.40.1. Please upgrade 🙂 Thank you guys for the feedback! And especially big thanks to @khamutov for implementing this in his PR! 🙂
